ttnn/ttnn/operations/pool.py

```diff
@@ -128,32 +133,8 @@ def _torch_global_avg_pool2d(input_tensor: ttnn.Tensor):
     return torch.nn.functional.global_avg_pool2d(input_tensor, output_size)
 
 
-def _global_avg_pool2d_validate_input_tensors(operation_name, input_tensor, *args, **kwargs):
-    ttnn.validate_input_tensor(
-        operation_name,
-        input_tensor,
-        ranks=(4,),
-        dtypes=(ttnn.bfloat16, ttnn.bfloat8_b, ttnn.uint16, ttnn.uint32),
-        layouts=(ttnn.TILE_LAYOUT,),
-        can_be_on_device=True,
-        can_be_on_cpu=False,
-    )
-
-
-@ttnn.register_operation(
-    name="ttnn.global_avg_pool2d",
-    validate_input_tensors=_global_avg_pool2d_validate_input_tensors,
-    golden_function=_torch_global_avg_pool2d,
+global_avg_pool2d = ttnn.register_operation(golden_function=_golden_function)(
+    ttnn._ttnn.operations.pool.global_avg_pool2d
 )
-def global_avg_pool2d(input_tensor: ttnn.Tensor, memory_config: ttnn.MemoryConfig = None) -> ttnn.Tensor:
-    r"""
-    Applies a 2D adaptive average pooling over an input signal composed of several input planes.
 
-    Arguments:
-        * :attr: input_tensor: the input tensor
-    """
-    if memory_config is None:
```

Comment:
behavior now is slightly different, if memory_config is none, we are taking it from the input tensor

Reply:
AFAIK, we should be doing DEFAULT_OUTPUT_MEMORY_CONFIG which is in dram interleaved. This is the non-optimal general path that should always work

Reply:
We have a thread discussing this. Likely will stay as is, but if needed, will change separately